src/peer_connection.cpp

```diff
@@ -5564,6 +5563,9 @@ namespace libtorrent
 		TORRENT_ASSERT(is_single_thread());
 		if (m_disconnecting) return;
 
+		// guard BEP17 and BEP19 seeds with depleted send buffer
+		if (type() != bittorrent_connection && m_send_buffer.empty()) return;
```

Comment:
could you elaborate a bit on this? If it makes sense to not request bandwidth for a web seed that doesn't have anything in the send buffer, wouldn't it make sense for normal bittorrent connections too?

Reply:
For regular bt connections it could also make sense if  `m_reading_bytes == 0'. I.e. if bittorrent connection has an empty send buffer and has no pending read operations it should be safe to skip bandwidth request as well.

Reply:
I was suggesting that this conditional could be simplified to:

``if (m_send_buffer.empty()) return;``